test/e2e/metrics_test.go

```diff
@@ -0,0 +1,176 @@
+)
+
+func setupMetricsEndpointTestCase(t *testing.T) (*framework.ClientSet, *operatorsv1.Console) {
+	client, _ := framework.StandardSetup(t)
+	routeForTest := tempRouteForTesting()
+	_, err := client.Routes.Routes(api.OpenShiftConsoleOperatorNamespace).Create(routeForTest)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("error: %s", err)
+	}
+	return client, nil
+}
+
+func cleanUpMetricsEndpointTestCase(t *testing.T, client *framework.ClientSet) {
+	routeForTest := tempRouteForTesting()
+	err := client.Routes.Routes(api.OpenShiftConsoleOperatorNamespace).Delete(routeForTest.Name, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("error: %s", err)
+	}
+	framework.StandardCleanup(t, client)
+}
+
+// This test essentially does the following to verify the `console_url` is in the metrics endpoint:
+// 	curl --insecure -H "Authorization: Bearer $(oc whoami --show-token)" $(oc get route metrics --template 'https://{{.spec.host}}/metrics') | grep console_url
+// alternatively:
+// 	oc exec -it console-operator-<pod-id> /bin/bash
+// 	curl --insecure -H "Authorization: Bearer <token>  https://localhost:8443/metrics | grep console_url
+func TestMetricsEndpoint(t *testing.T) {
+	client, _ := setupMetricsEndpointTestCase(t)
+	defer cleanUpMetricsEndpointTestCase(t, client)
+
+	metricsURL := getMetricsURL(t, client)
+	fmt.Printf("fetching metrics url.... %v\n", metricsURL)
+
+	fmt.Printf("making request...\n")
+	respString := metricsRequest(t, metricsURL)
+
+	toFind := "console_url"
+	fmt.Printf("searching for %v\n", toFind)
+	found := findLineInResponse(respString, toFind)
+	if !found {
+		t.Fatalf("did not find %v", toFind)
+	}
+}
+
+func findLineInResponse(haystack, needle string) (found bool) {
+	scanner := bufio.NewScanner(strings.NewReader(haystack))
+	for scanner.Scan() {
+		found := strings.Contains(scanner.Text(), needle)
+		if found {
+			fmt.Printf("found %v\n", scanner.Text())
+			return true
+		}
+	}
+	return false
+}
+
+func metricsRequest(t *testing.T, routeForMetrics string) string {
+	bearer := getBearerToken(t)
+	req := getRequest(t, routeForMetrics, bearer)
+	insecureClient := getInsecureClient()
+
+	resp, err := insecureClient.Do(req)
+	if err != nil {
+		t.Fatalf("error: %s", err)
+	}
+	defer resp.Body.Close()
+
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("error: %s", err)
+	}
+
+	return string(bytes)
+}
+
+// a request with a route & the bearer token
+func getRequest(t *testing.T, metricsURL, bearer string) *http.Request {
+	req, err := http.NewRequest("GET", metricsURL, nil)
+	if err != nil {
+
+	}
+	req.Header.Add("Authorization", bearer)
+	return req
+}
+
+func getInsecureClient() *http.Client {
+	// ignore self signed certs for testing purposes
+	insecureTransport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+	insecureClient := &http.Client{
+		Transport: insecureTransport,
+	}
+	return insecureClient
```

Comment:
lets make it a single struct
```suggestion
	return &http.Client{
		Transport: http.Transport{
			TLSClientConfig: tls.Config{
				InsecureSkipVerify: true,
			},
		},
	}
```

Reply:
and `return` immediately, done.